File: Java/Easy/Accepted/next-greater-element-i_6.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        
        int[] ans = new int[nums1.length];
        for (int i = 0; i < nums1.length; i++) {
            ans[i] = -1;
            int j;
            for (j = 0; j < nums2.length; j++) {
                if (nums2[j] == nums1[i]) break;
            }
            for (int k = j + 1; k < nums2.length; k++) {
                if (nums2[k] > nums1[i]) {
                    ans[i] = nums2[k];
                    break;
                }
            }
        }
        return ans;
    }
}

File: Rust/Easy/Accepted/next-greater-element-i_6.rs
impl Solution {
    pub fn next_greater_element(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        
        let mut res = vec![];
        for i in nums1 {
            let mut j = nums2.iter().position(|&x| x == i).unwrap();
            while j < nums2.len() {
                if nums2[j] > i {
                    res.push(nums2[j]);
                    break;
                }
                j += 1;
            }
            if j == nums2.len() {
                res.push(-1);
            }
        }
        res
    }
}